app/src/main/java/com/rosario/covid19/viewModel/HomeViewModel.kt

```diff
@@ -1,107 +1,36 @@
-import com.rosario.covid19.R
+import androidx.lifecycle.*
 import com.rosario.covid19.data.model.DataResponse
-import com.rosario.covid19.data.model.Report
 import com.rosario.covid19.domain.ReportUseCase
-import io.reactivex.android.schedulers.AndroidSchedulers
-import io.reactivex.disposables.Disposable
-import io.reactivex.schedulers.Schedulers
+import com.rosario.covid19.util.Resource
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 /**
  * Handle connection to server and show data to UI
  */
 class HomeViewModel @Inject constructor(
-    private val reportUseCase: ReportUseCase,
-    private val context: Context
-) {
+    private val reportUseCase: ReportUseCase
+) : ViewModel() {
 
-    private lateinit var subscription: Disposable
-    var response = MutableLiveData<Report>()
-    var error = MutableLiveData<Throwable>()
-    var progressBar = MutableLiveData<Int>().apply { postValue(View.GONE) }
-    var dataVisibility = MutableLiveData<Int>().apply { postValue(View.GONE) }
     var confirmed = MutableLiveData<String>().apply { postValue("") }
     var deaths = MutableLiveData<String>().apply { postValue("") }
-
+    var reportMutableLiveData =
+        MutableLiveData<Resource<DataResponse>>().apply { postValue(Resource.loading(data = null)) }
 
     /**
      *  Get data from server
      * @param date
      */
     fun getReport(date: String) {
-        subscription = reportUseCase.getReport(date)
-            .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
-            .doOnSubscribe { progressStart() }
-            .doOnTerminate { hideProgress() }
-            .subscribe({ result -> registerSuccess(result) },
-                { error -> onError(error) })
-    }
-
-    /**
-     *  Hide progress bar in UI
-     */
-    private fun hideProgress() {
-        progressBar.postValue(View.GONE)
-    }
-
-    /**
-     *  Show progress bar to UI
-     */
-    private fun progressStart() {
-        progressBar.postValue(View.VISIBLE)
-    }
-
-    /**
-     *  Get success response data
-     */
-    private fun registerSuccess(result: DataResponse?) {
-        if (result!!.data.size > 0) {
-            if (result.data.containsKey("confirmed")) {
-                val dataConfirmed  = result.data["confirmed"] as Double
-                val dataDeaths = result.data["deaths"] as Double
-
-                confirmed.postValue(
-                    String.format(
-                        context.resources.getString(R.string.cases_confirmed),
-                        dataConfirmed.toInt()
-                    )
-                )
-                deaths.postValue(
-                    String.format(
-                        context.resources.getString(R.string.deaths_confirmed),
-                        dataDeaths.toInt()
-                    )
-                )
+        viewModelScope.launch {
+            reportMutableLiveData.postValue(Resource.loading(data = null))
```

Comment:
Will you use the data passed to _loading_ function? if not, then you can remove that parameter from the _loading_ function  instead of pass always _null_